library/src/main/java/com/getstream/sdk/chat/viewmodel/ChannelViewModel.java

```diff
@@ -18,18 +18,21 @@
 import com.getstream.sdk.chat.enums.Pagination;
 import com.getstream.sdk.chat.model.Channel;
 import com.getstream.sdk.chat.model.Event;
+import com.getstream.sdk.chat.model.ModelType;
```

Comment:
so if I understand correctly, we are using this view model to render both channels and a thread; not sure if this is a good idea, perhaps better to create a new ViewModel, the feature-set of threads and message lists is quite different (ie. typing, mark read, actions are different) and using 1 class to do both makes the VM more complex.

@tschellenbach what do you think? Perhaps extract an AbstractClass for message list and create 2 implementations?

Reply:
If we are using another ViewModel for Thread, we will have the other problem. How to handle Interaction such as Edit message, Reaction,...

Please see MoreActionDialog, while creating new instance for it, we must transfer viewModel.

Reply:
1 viewmodel approach seems correct.